build-tools/scripts/RunTests.targets

```diff
@@ -0,0 +1,89 @@
+    <_TestAssembly Include="$(TestAssembly)" Condition=" '$(TestAssembly)' != '' " />
+    <_JavaInteropTestResults Include="$(JavaInteropSourceDirectory)\TestResult-*.xml" />
+    <_ApkTestProject Include="$(_TopDir)\src\Mono.Android\Test\Mono.Android-Tests.csproj" />
+    <_ApkTestProject Include="$(_TopDir)\tests\CodeGen-Binding\Xamarin.Android.JcwGen-Tests\Xamarin.Android.JcwGen-Tests.csproj" />
+    <_ApkTestProject Include="$(_TopDir)\tests\locales\Xamarin.Android.Locale-Tests\Xamarin.Android.Locale-Tests.csproj" />
+    <_ApkTestProject Include="$(_TopDir)\tests\Xamarin.Android.Bcl-Tests\Xamarin.Android.Bcl-Tests.csproj" />
+    <_ApkTestProject Include="$(_TopDir)\tests\Xamarin.Forms-Performance-Integration\Droid\Xamarin.Forms.Performance.Integration.Droid.csproj" />
+    <_ApkTestProjectAot Include="$(_TopDir)\src\Mono.Android\Test\Mono.Android-Tests.csproj" />
+  </ItemGroup>
+  <Target Name="RunNUnitTests">
+    <Exec
+        Command="$(_NUnit) $(NUNIT_EXTRA) %(_TestAssembly.Identity) $(_Test) --result=&quot;TestResult-%(Filename).xml;format=nunit2&quot; --output=&quot;bin\Test$(Configuration)\TestOutput-%(Filename).txt&quot;"
+        WorkingDirectory="$(_TopDir)"
+        ContinueOnError="True"
+    />
+    <ItemGroup>
+      <_RenameTestCasesGlob Include="$(_TopDir)\TestResult-*Tests.xml" />
+    </ItemGroup>
+    <PropertyGroup>
+      <_RenamedTestCases>@(_RenameTestCasesGlob)</_RenamedTestCases>
+    </PropertyGroup>
+    <MSBuild
+        Projects="$(MSBuildThisFileDirectory)TestApks.targets"
+        Targets="RenameTestCases"
+        Properties="Configuration=$(Configuration);RenameTestCasesGlob=$(_RenamedTestCases)"
+    />
+  </Target>
+  <Target Name="RunJavaInteropTests">
+    <MSBuild Projects="$(JavaInteropSourceDirectory)\Java.Interop.sln" Condition=" '$(HostOS)' == 'Windows' " />
+    <Exec
+        Command ="make -C &quot;$(JavaInteropSourceDirectory)&quot; CONFIGURATION=$(Configuration) all"
+        Condition=" '$(HostOS)' != 'Windows' "
+    />
+    <SetEnvironmentVariable Name="ANDROID_SDK_PATH" Value="$(AndroidSdkFullPath)" />
+    <MSBuild 
+        Projects="$(JavaInteropSourceDirectory)\build-tools\scripts\RunNUnitTests.targets"
+        Condition=" '$(HostOS)' == 'Windows' "
+    />
+    <Exec
+        Command ="make -C &quot;$(JavaInteropSourceDirectory)&quot; CONFIGURATION=$(Configuration) run-all-tests || true"
+        Condition=" '$(HostOS)' != 'Windows' "
+    />
+    <ItemGroup>
+      <_RenameTestCasesGlob Include="$(JavaInteropSourceDirectory)\TestResult-*Tests.xml" />
+    </ItemGroup>
+    <PropertyGroup>
+      <_RenamedTestCases>@(_RenameTestCasesGlob)</_RenamedTestCases>
+    </PropertyGroup>
+    <MSBuild
+        Projects="$(MSBuildThisFileDirectory)TestApks.targets"
+        Targets="RenameTestCases"
+        Properties="Configuration=$(Configuration);RenameTestCasesGlob=$(_RenamedTestCases)"
```

Comment:
Does this actually work? `Properties` is a `;`-delimited list of properties, and `$(_RenamedTestCases)` may in turn contain `;`s. If that works....*wow*.

Reply:
This _does_ seem to work, even under `xbuild`. I will need to read the results of the Jenkins build carefully to make sure it looks the same as older builds.